src/types.rs

```diff
@@ -162,23 +163,33 @@ fn field_name_to_sql_with_alias() {
 // to do operations. The AST is much more
 // sophisticated and hence can become cumbersome
 #[derive(Serialize, Clone, Debug)]
-pub enum GQLArgType {
+// NOTE: We can keep increasing the types that this
+// struct uses to ensure that we are writing type safe code.
+// Certainly, there's no element of elegance to it. But,
+// nevertheless should help the developer to write much simpler
+// functions for the processing of the AST and SQL code
+pub enum GQLArgType<T> {
     // NOTE: supported for [distinct_on]
     String(String),
     // NOTE: supported for [limit, offset]
     Int(i64),
-    // Float(f32)
+    // NOTE: supported for [order_by]
+    Object(IndexMap<String, T>),
 }
```

Comment:
I'm a bit worried that this enum field doesn't clearly show what they are for. can we consider sth like this?
```rust
pub enum GQLArgType<T> {
    // NOTE: supported for [distinct_on]
    DistinctOn(String),
    // NOTE: supported for [limit, offset]
    Limit(i64),
    Offset(i64),
    // NOTE: supported for [order_by]
    OrderBy(IndexMap<String, T>),
}
```

Reply:
I am in favour of this, but this refactor can be picked up at a later stage, I presume? Keeping it argument agnostic, I think we can re-use a lot more code. Plus, there's only one more `argtype` that will be added for queries, i.e `where`.

Reply:
I'll be merging this PR now, but yeah, you could explain your reasoning too and we can take a more informed decision upon that.